src/variable.cpp

```diff
@@ -118,4 +131,42 @@ std::string Variable::getInitialValue() const
     return mPimpl->mInitialValue;
 }
 
+void Variable::setInterfaceType(const Variable::INTERFACE_TYPES interfaceType)
+{
+    mPimpl->mInterfaceType = interfaceType;
+}
+
+Variable::INTERFACE_TYPES Variable::getInterfaceType() const
+{
+    return mPimpl->mInterfaceType;
+}
+
+std::string interfaceTypeToString(Variable::INTERFACE_TYPES interfaceType)
+{
+    std::string str;
+    switch (interfaceType) {
+    case Variable::INTERFACE_TYPES::INTERFACE_TYPE_UNDEFINED: {
+        str = "";
+        break;
+    }
+    case Variable::INTERFACE_TYPES::INTERFACE_TYPE_NONE: {
+        str = "none";
+        break;
```

Comment:
Hmm, good point, I have indeed seen that ```control reaches end of non-void function``` compile error before (only with MSVC I believe), even though everything would have actually been fine here.